packages/optimistic-oracle/src/proposer.js

```diff
@@ -175,8 +175,126 @@ class OptimisticOracleProposer {
+      if (!priceFeed) {
+        this.logger.error({
+          at: "OptimisticOracleProposer#sendDisputes",
+          message: "Failed to construct a PriceFeed for price request",
+          priceRequest
+        });
+        continue;
+      }
+
+      // With pricefeed successfully constructed, confirm the proposal price
+      await priceFeed.update();
+      let disputePrice;
+      try {
+        disputePrice = (await priceFeed.getHistoricalPrice(priceRequest.timestamp)).toString();
+      } catch (error) {
+        this.logger.error({
+          at: "OptimisticOracleProposer#sendDisputes",
+          message: "Failed to query historical price for price request",
+          disputePrice,
+          error
+        });
+        continue;
+      }
+
+      // If proposal price is not equal to the dispute price, then
+      // prepare dispute.
+      // TODO: Implement dispute buffer feature to only dispute if proposal and dispute price
+      // differ using some margin of error. For example, we could define a variable `proposalErrorPrecision`
+      // and set it to 5, to only send disputes if prices differ after rounding to 5 decimals of precision.
+      let isPriceDisputable = !this.toBN(disputePrice).eq(this.toBN(proposalPrice));
+      if (isPriceDisputable) {
+        // Create the transaction.
+        const dispute = this.ooContract.methods.disputePrice(
+          priceRequest.requester,
+          this.utf8ToHex(priceRequest.identifier),
+          priceRequest.timestamp,
+          priceRequest.ancillaryData
+        );
+
+        // Simple version of inventory management: simulate the transaction and assume that if it fails,
+        // the caller didn't have enough collateral.
+        let disputeBond, gasEstimation;
+        try {
+          [disputeBond, gasEstimation] = await Promise.all([
+            dispute.call({ from: this.account }),
+            dispute.estimateGas({ from: this.account })
+          ]);
+        } catch (error) {
+          this.logger.error({
+            at: "OptimisticOracleProposer#sendDisputes",
+            message: "Cannot dispute price: not enough collateral (or large enough approval)✋",
+            proposer: this.account,
+            disputeBond,
+            priceRequest,
+            error
+          });
+          continue;
+        }
+        const txnConfig = {
+          from: this.account,
+          gas: Math.min(Math.floor(gasEstimation * this.GAS_LIMIT_BUFFER), this.txnGasLimit),
+          gasPrice: this.gasEstimator.getCurrentFastPrice()
+        };
+
+        this.logger.debug({
+          at: "OptimisticOracleProposer#sendDisputes",
+          message: "Disputing proposal",
+          priceRequest,
+          proposalPrice,
+          disputePrice,
+          txnConfig
+        });
+
+        // Send the transaction or report failure.
+        let receipt;
+        try {
+          receipt = await dispute.send(txnConfig);
```

Comment:
do you not want to use `ynatm`?

Reply:
I think we should deprecate using `ynatm` since we don't use it in the `disputer`, and its easier to not have to maintain an external repo